src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -512,6 +490,107 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
             return actualArguments.AsImmutableOrNull();
         }
 
+        internal static ImmutableArray<BoundExpression> MakeArgumentsInEvaluationOrder(
+            Binder binder,
+            SyntaxNode syntax,
+            ImmutableArray<BoundExpression> arguments,
+            Symbol methodOrIndexer,
+            MethodSymbol optionalParametersMethod,
+            bool expanded,
+            ImmutableArray<int> argsToParamsOpt,
+            bool invokedAsExtensionMethod,
+            /* out */ ArrayBuilder<ArgumentKind> argumentKindBuilder,
+            /* out */ ArrayBuilder<ParameterSymbol> parameterSymbolBuilder,
+            ThreeState enableCallerInfo = ThreeState.Unknown)
+        {
+            Debug.Assert(binder != null && argumentKindBuilder.IsEmpty() && parameterSymbolBuilder.IsEmpty());
+
+            // Either the methodOrIndexer is a property, in which case the method used
+            // for optional parameters is an accessor of that property (or an overridden
+            // property), or the methodOrIndexer is used for optional parameters directly.
+            Debug.Assert(((methodOrIndexer.Kind == SymbolKind.Property) && optionalParametersMethod.IsAccessor()) ||
```

Comment:
Well, even though the comment here says so, it turns out there's a bug in my code that `optionalParametersMethod` is never used for optional parameters (or anywhere). So the parameter symbol for default value is indeed corresponding to property symbol instead of getter/setter symbol.

As a result of this, we don't need to distinguish whether the indexer access is through setter or getter because `optionalParametersMethod` parameter can be removed from `MakeArgumentsInEvaluationOrder`.

I do think it make s more sense to distinguish indexer/property access through setter and getter, we probably should discuss this in our next design meeting. I will file an issue for that.